pvm_2_videos.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime
+import os
+from pythonosc import dispatcher, osc_server
+import argparse
+from omxplayer.player import OMXPlayer
+import logging
+from logging.handlers import TimedRotatingFileHandler
+import sys
+
+def _init_logger():
+	logger = logging.getLogger("PVM")
+	logger.setLevel(logging.INFO)
+	handler = logging.StreamHandler(sys.stderr)
+	fileHandler = TimedRotatingFileHandler('./log/{:%Y-%m-%d %H:%M:%S}.log'.format(datetime.now()),  when='midnight')
+	handler.setLevel(logging.INFO)
+	formatter = logging.Formatter("%(asctime)s;%(levelname)s;%(message)s",
+                              "%Y-%m-%d %H:%M:%S")
+	fileHandler.setFormatter(formatter)
+	fileHandler.suffix = '%Y_%m_%d.log'
+	logger.addHandler(fileHandler)
+	handler.setFormatter(formatter)
+	logger.addHandler(handler)
+
+_init_logger()
+_logger = logging.getLogger("PVM")
+_logger.info("Logging system initilized in %s", os.getcwd())
+
+# Place your videos in this folder for autostart
+PEFIX_PATH = "/home/pi/Videos/"
+VIDEO_PATH_ONE = "jellyfish720p.mp4"
+VIDEO_PATH_TWO = "jellyfish720p.mp4"
+media1 = ""
+media2 = ""
+IS_FILE_SET = False
+
+def parse_commands(*args):
+	global media1
+	global media2
+	global VIDEO_PATH_ONE
+	global VIDEO_PATH_TWO
+	global IS_FILE_SET
+	command = args[1]
+	_logger.info("Received command: %s", command)
+	if len(args)>2:
+		value = args[2]
+		_logger.info("Received command: %s", str(value))
+		pass
+	# TODO: Create another python file to control two display
+	if command=="file":
+		if "|" not in value:
+			_logger.info("Please pass two video name like 1.mp4,2.mp4.")
+			return
+		VIDEO_PATH_ONE, VIDEO_PATH_TWO = value.split("|")
+		_logger.info("File set: %s, %s", PEFIX_PATH + VIDEO_PATH_ONE, PEFIX_PATH + VIDEO_PATH_TWO)
+		IS_FILE_SET = True
+		media1 = OMXPlayer(PEFIX_PATH + VIDEO_PATH_ONE, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=['--loop', '--display', '2'])
+		media1.pause()
+		media2 = OMXPlayer(PEFIX_PATH + VIDEO_PATH_TWO, dbus_name='org.mpris.MediaPlayer2.omxplayer2', args=['--loop', '--display', '7'])
+		media2.pause()
+		return
+
+	if not IS_FILE_SET:
+		_logger.info("Command %s failed because of the file is unset.", command)
+		return
+
+	if command=="start":
+		if media1.can_play():
+			# check two videos' position
+			pos1=media1.position()
+			pos2=media2.position()
+			offset =abs(pos1-pos2)
+			if(offset>1): #more than 1 second offset, direct jump to correct position
+				media2.set_position(pos1)
```

Comment:
Interesting hack to try to keep apparently in sync, but this will not work:
- our goal is to be able to have things in sync to the frame, not somewhere around the same second (that could be 30 frames!)
- at some point there will be intentional offsets, as you can see in the examples folder. Meaning that if you hardcode this here you will be removing that possibility.

Reply:
Let me test what's the value of pos1. And then get back to this comment.
If the value of pos1 is like 1.123, we can use the set_position function to sync videos in the same frame.

Reply:
Fixed.